scene/resources/visual_shader.cpp

```diff
@@ -3313,6 +3314,14 @@ String VisualShaderNodeUniformRef::get_output_port_name(int p_port) const {
 	return "";
 }
 
+void VisualShaderNodeUniformRef::set_shader(const Ref<VisualShader> &p_shader) {
+	shader = p_shader;
+}
+
+Ref<VisualShader> VisualShaderNodeUniformRef::get_shader() const {
+	return shader;
+}
```

Comment:
I'm not an expert but since you only seem to need the RID, I think it might be better to just save the RID itself instead of a Ref to the shader. Would be good to confirm with @reduz, this is just a gut feeling from me.

Reply:
Changed to RID...